subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/VersionConstraintInternal.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.api.internal.artifacts;
+
+import org.gradle.api.artifacts.VersionConstraint;
+import org.gradle.api.internal.artifacts.ivyservice.ivyresolve.strategy.VersionSelector;
+import org.gradle.api.internal.artifacts.ivyservice.ivyresolve.strategy.VersionSelectorScheme;
+
+public interface VersionConstraintInternal extends VersionConstraint {
```

Comment:
It would be great if we could supply the `VersionSelectorScheme` when constructing the `VersionConstraint`, rather than each time it is used. This will always be a `DefaultVersionSelectorScheme`: the only time we use a `MavenVersionSelectorScheme` is to when we are reading or writing a POM file, and it's only used to convert version strings "latest.release" <-> "RELEASE" and "latest.integration" <-> "LATEST".

I guess this might be tricky because we need to be able to create `ModuleVersionSelector` instances without too much ceremony...